contrib/weights-from-festa-sommariva.py

```diff
@@ -0,0 +1,116 @@
+r"""
+The associated paper is given at
+
+    M. Festa, A. Sommariva,
+    Computing Almost Minimal Formulas on the Square,
+    Journal of Computational and Applied Mathematics, Vol. 236, pp. 4296--4302, 2012,
+    https://doi.org/10.1016/j.cam.2012.05.021
+
+The quadrature rules extracted here can be found
+    https://www.math.unipd.it/~alvise/sets.html
+
+These rules are licensed under the GPL and are not included by default.
+"""
+
+import re
+
+import numpy as np
+from pytools import download_from_web_if_not_present
+
+
+_re_degree = re.compile(r"DEGREE:\s+(\d+)")
+_re_xw = re.compile(r"xw\s?=\s?\[(.+?)\];", re.DOTALL)
+
+_PYTHON_TEMPLATE = """# GENERATED by modepy/contrib/weights-from-festa-sommariva.py
+# DO NOT EDIT
+
+import numpy as np
+from modepy.quadrature import Quadrature, QuadratureRuleUnavailable
+
+
+class FestaSommarivaQuadrature(Quadrature):
+    def __init__(self, order):
+        try:
+            rule = square_data[order]
+        except KeyError:
+            raise QuadratureRuleUnavailable
+
+        super().__init__(rule["points"], rule["weights"])
+        self.exact_to = rule["quad_degree"]
+
+
+def process_rules(rules):
+    result = {}
+    for order, rule in rules.items():
+        result[order] = {
+            k: np.array(v) if isinstance(v, list) else v
+            for k, v in rule.items()
+            }
+
+    return result
+
+
+square_data = process_rules(%s)"""
+
+
+def generate_festa_sommariva_quadrature_rules(outfile, pointset="set_amr_square"):
+    filename = f"{pointset}.m"
+    download_from_web_if_not_present(
+            url=f"https://www.math.unipd.it/~alvise/POINTSETS/{filename}",
```

Comment:
Ask https://archive.org to archive this and use the resulting URL.

Reply:
Fixed in 1e73044 (hopefully that's what you meant)